optimum/exporters/tasks.py

```diff
@@ -912,38 +912,37 @@ def determine_framework(
         if framework is not None:
             return framework
 
-        framework_map = {"pt": "PyTorch", "tf": "TensorFlow"}
-
         full_model_path = Path(model_name_or_path) / subfolder
         if full_model_path.is_dir():
-            if (full_model_path / WEIGHTS_NAME).is_file():
-                framework = "pt"
-            elif (full_model_path / TF2_WEIGHTS_NAME).is_file():
-                framework = "tf"
-            else:
-                raise FileNotFoundError(
-                    "Cannot determine framework from given checkpoint location."
-                    f" There should be a {WEIGHTS_NAME} for PyTorch"
-                    f" or {TF2_WEIGHTS_NAME} for TensorFlow."
-                )
-            logger.info(f"Local {framework_map[framework]} model found.")
+            all_files = [
+                os.path.relpath(os.path.join(dirpath, file), full_model_path)
+                for dirpath, _, filenames in os.walk(full_model_path)
+                for file in filenames
+            ]
         else:
             if not isinstance(model_name_or_path, str):
                 model_name_or_path = str(model_name_or_path)
-            try:
-                url = huggingface_hub.hf_hub_url(model_name_or_path, WEIGHTS_NAME, subfolder=subfolder)
-                huggingface_hub.get_hf_file_metadata(url)
-                framework = "pt"
-            except Exception:
-                pass
-
-            if framework is None:
-                try:
-                    url = huggingface_hub.hf_hub_url(model_name_or_path, TF2_WEIGHTS_NAME, subfolder=subfolder)
-                    huggingface_hub.get_hf_file_metadata(url)
-                    framework = "tf"
-                except Exception:
-                    pass
+            all_files = huggingface_hub.list_repo_files(model_name_or_path, repo_type="model")
+            all_files = [file for file in all_files if file.startswith(subfolder)]
+
+        weight_name = Path(WEIGHTS_NAME).stem
+        weight_extension = Path(WEIGHTS_NAME).suffix
+        is_pt_weight_file = (file.startswith(weight_name) and file.endswith(weight_extension) for file in all_files)
+
+        weight_name = Path(TF2_WEIGHTS_NAME).stem
+        weight_extension = Path(TF2_WEIGHTS_NAME).suffix
+        is_tf_weight_file = (file.startswith(weight_name) and file.endswith(weight_extension) for file in all_files)
+
+        if any(is_pt_weight_file):
+            framework = "pt"
+        elif any(is_tf_weight_file):
+            framework = "tf"
+        else:
+            raise FileNotFoundError(
+                "Cannot determine framework from given checkpoint location."
+                f" There should be a {Path(WEIGHTS_NAME).stem}*{Path(WEIGHTS_NAME).suffix} for PyTorch"
+                f" or {Path(TF2_WEIGHTS_NAME).stem}*{Path(TF2_WEIGHTS_NAME).suffix} for TensorFlow."
+            )
```

Comment:
I think we should log which framework has been chosen, something like:

```
logger.info(f"No framework specified so {framework} has been automatically chosen.")
```

Because here PT will have the priority over TF when there are checkpoints for both frameworks in the repo (which is fine), like BERT for instance: https://huggingface.co/bert-base-uncased/tree/main

Reply:
This is already logged in the CLI: `Framework not specified. Using pt to export to ONNX.` If passing `export=True` this is indeed not logged.